toolchain/check/handle_interface.cpp

```diff
@@ -38,7 +38,7 @@ static auto BuildInterfaceDecl(Context& context,
 
   // Process modifiers.
   CheckAccessModifiersOnDecl(context, Lex::TokenKind::Interface,
-                             name_context.target_scope_id);
+                             name_context.enclosing_scope_id);
```

Comment:
I think this is wrong (pre-existing bug), because `enclosing_scope_id` refers to the resolved prefix of the name, which isn't necessarily the enclosing scope. For example, for 
```carbon
class C { 
  protected interface I {}    
  protected interface I;  
}
```
... we currently reject the `protected` on the second declaration of `I`:
```console
/proc/self/fd/11:4:3: ERROR: `protected` not allowed on `interface` declaration, `protected` is only allowed on class members.
  protected interface I;
  ^~~~~~~~~
```
... because `name_context.enclosing_scope_id` is resolved to the interface declaration here, and so we're incorrectly checking this interface declaration as if it appeared within itself.

(The above example would be invalid for other reasons -- it's redeclaring `I` after a definition -- but I think the point remains that the current name and the new name don't really convey what this field means.)

Reply:
Note, while I agree it seems suspicious, this PR is not changing behavior.